src/Foundatio/Queues/QueueBase.cs

```diff
@@ -53,6 +56,10 @@ protected QueueBase(TOptions options) : base(options?.TimeProvider, options?.Log
 
         _queueDisposedCancellationTokenSource = new CancellationTokenSource();
 
+        var resiliencePipelineProvider = _options.GetResiliencePolicyProvider();
+        _abandonPolicy = resiliencePipelineProvider?.GetPolicy(nameof(IQueue<T>.AbandonAsync)) ?? new ResiliencePolicy(_timeProvider, _logger) { MaxAttempts = 3, RetryInterval = TimeSpan.Zero };
```

Comment:
how does the nameof here work on the generic type? What if you wanted a policy per queue type. e.g., priority customers get 5x retries where free customers get 1 retry or something like that (bad example). Or media processing gets 1 retry but payments get x.

Reply:
It now looks for `QueueBase<T, TOptions>, IQueue<T>, IQueue` in order. So if any policies are registered for those, it will use them. So you can no define policies for specific queue types.